app/controllers/application_controller.rb

```diff
@@ -62,7 +62,9 @@ def redirect_to(options = {}, response_status = {})
   end
 
   def set_checkout_redirect
-    return unless URI(request.referer.to_s).path == main_app.checkout_path
+    referer_path = URI(request.referer.to_s).path
+    return unless referer_path == main_app.checkout_path ||
+                  referer_path == main_app.checkout_step_path(:details)
 
     session["spree_user_return_to"] = main_app.checkout_path
   end
```

Comment:
I think that this method shouldn't be in the ApplicationController but I haven't looked through all the code to suggest a better way.

Reply:
I only changed the method behavior, but I did not move it. 
Why do you suggest this one should not be here? I can see there also there are some others methods in `application_controller` that refers to _redirect_, like `redirect_to` or `after_sign_in_path_for`. 
Just to be sure to understand the reasons: then I can start looking for a better place!

Reply:
Sorry, I wasn't criticising you. I know that you didn't move it there. This method seems to be checkout related while the application controller contains common code used by most other controllers. A generic redirect is very common in most controllers. But `set_checkout_redirect` is only used when logging in. Only two controllers use it.

I would probably leave this until the split checkout is complete and active. Then we can have some new logic that either returns the user to where they came from (referer) or looks at the order state to go to the right shopping/checkout step. But doing that with a feature toggle sounds too complicated.

Reply:
> Sorry, I wasn't criticising you.

I wasn't complaining, sorry @mkllnk ! That was just to be sure to understand.

> I would probably leave this until the split checkout is complete and active.

I agree. Let's resolve this conversation then?